lettuce/simulation.py

```diff
@@ -50,7 +50,8 @@ def __init__(self, flow, lattice, collision, streaming):
         self.no_collision_mask = lattice.convert_to_tensor(self.no_collision_mask)
         for boundary in self.flow.boundaries:
             if boundary.__class__.__name__ == "BounceBackBoundary":
-                self.no_collision_mask = boundary.mask | self.no_collision_mask
+                self.no_collision_mask = boundary.mask.bool() | self.no_collision_mask.bool()
+                self.no_collision_mask = lattice.convert_to_tensor(self.no_collision_mask).byte()
```

Comment:
Why do this with bool and byte?

Reply:
Apparently since pytorch 1.2 (lettuce seems to be on 1.4) pytorch also includes a bool tensor, which torch.where ect. can work with, so the byte()-conversion for unit8 is no longer necessary.
I've swapped out the uint8 related stuff against bool now, so we can save a bit of memory and make the function of the masks more obvious. Also the masks are now bool by default, so the logical operation does not need any conversion before hand.

Reply:
Apparently since pytorch 1.2 (lettuce seems to be on 1.4) pytorch also includes a bool tensor, which torch.where ect. can work with, so the byte()-conversion for unit8 is no longer necessary.
I've swapped out the uint8 related stuff against bool now, so we can save a bit of memory and make the function of the masks more obvious. Also the masks are now bool by default, so the logical operation does not need any conversion before hand.